sncosmo/spectral.py

```diff
@@ -468,16 +500,89 @@ def zpbandflux(self, band):
 
         return bandflux
 
-    def band_flux_to_mag(self, flux, band):
-        """Convert flux (photons / s / cm^2) to magnitude."""
-        return -2.5 * math.log10(flux / self.zpbandflux(band))
+class NaturalMagSystem(MagSystem):
 
-    def band_mag_to_flux(self, mag, band):
-        """Convert magnitude to flux in photons / s / cm^2"""
-        return self.zpbandflux(band) * 10.**(-0.4 * mag)
+    """A magnitude system in which the magnitudes / fluxes are
+    calibrated such that they represent what is measured through the
+    filters attached to a specific telescope (not standard filters). 
+
+    Parameters
+    ----------
+    
+    bands: iterable of `~sncosmo.Bandpass` or str
+        The filters in the magnitude system.
+    zeropoints: `~numpy.ndarray`
+        The zeropoints of the filters. 
+    standards: iterable of `~sncosmo.Spectrum`,
+        The spectrophotmetric flux standards for each band.
+    """
+
+    def __init__(self, bands, zeropoints, standards, name=None):
+        super(NaturalMagSystem, self).__init__(name)
+        if len(bands) != len(zeropoints):
+            raise ValueError, 'Number of passed bands (%d) does not equal '  \
+                'the number of passed zeropoints (%d).'  % (len(bands),      
+                                                            len(zeropoints))
+        if len(standards) != len(zeropoints):
+            raise ValueError, 'Number of passed stanadards (%d) does '       \
+                ' not equal the number of passed '                           \
+                'zeropoints (%d).' % (len(standards),      
+                                      len(zeropoints))
+
+        bandnames = [get_bandpass(b).name for b in bands]
+        self._bands = bandnames
+        self._zeropoints = dict(zip(bandnames, zeropoints))
+        self._standards = dict(zip(bandnames, standards))
+
+    @_verify_band(0)
+    def zpbandflux(self, band):
+        """Flux of an object with magnitude zero in the given bandpass.
+
+        Parameters
+        ----------
+        bandpass : `~sncosmo.spectral.Bandpass` or str
 
+        Returns
+        -------
+        bandflux : float
+            Flux in photons / s / cm^2.
+        """
+        return 10**(0.4 * self.zeropoints[band])
+    
+    @property
+    def zeropoints(self):
+        return self._zeropoints
 
-class SpectralMagSystem(MagSystem):
+    @property
+    def bands(self):
+        return self._bands
+    
+    @property
+    def standards(self):
+        return self._standards
+
+    @_verify_band(0)
+    def _refspectrum_bandflux(self, band):
+        return self.standards[band].bandflux(band)
```

Comment:
I think we could forgo the use of a decorator here, and just use a normal method. This is just as compact:

```python
def _refspectrum_bandflux(self, band):
    self._verify_band(band)
    return self.standards[band].bandflux(band)
```
and easier to parse.